src/content/docs/en/guides/backend/turso.mdx

```diff
@@ -0,0 +1,126 @@
+service: Turso
+stub: false
+i18nReady: true
+---
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+[Turso](https://turso.tech) is a distributed database built on libSQL, a fork of SQLite. It is optimized for low query latency, making it suitable for global applications.
+
+## Initializing Turso in Astro
+
+### Prerequisites
+
+- The [Turso CLI](https://docs.turso.tech/reference/turso-cli) installed and signed in
+- A [Turso](https://turso.tech) Database with schema
+- Your Database URL
+- An Access Token
+
+### Configure environment variables
+
+Obtain your database URL using the following command:
+
+```bash
+turso db show <database-name> --url
+```
+
+Create an auth token for the database:
+
+```bash
+turso db tokens create <database-name>
+```
+
+Add the output from both commands above into your `.env` file at the root of your project. If this file does not exist, create one.
+
+```ini title=".env"
+TURSO_DATABASE_URL=libsql://...
+TURSO_AUTH_TOKEN=
+```
+
+:::caution
+Do not use the `PUBLIC_` prefix when creating these private [environment variables](/en/guides/environment-variables/). This will expose these values on the client.
+:::
+
+### Install LibSQL Client
+
+Install the `@libsql/client` to connect Turso to Astro:
+
+<PackageManagerTabs>
+  <Fragment slot="npm">
+    ```shell
+    npm install @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="pnpm">
+    ```shell
+    pnpm add @libsql/client
+    ```
+  </Fragment>
+  <Fragment slot="yarn">
+    ```shell
+    yarn add @libsql/client
+    ```
+  </Fragment>
+</PackageManagerTabs>
+
+### Initialize a new client
+
+Create a file `turso.ts` in the `src` folder and invoke `createClient`, passing it `TURSO_DATABASE_URL` and `TURSO_AUTH_TOKEN`:
+
+```ts title="src/turso.ts"
+import { createClient } from "@libsql/client/web";
+
+export const turso = createClient({
+  url: import.meta.env.TURSO_DATABASE_URL,
+  authToken: import.meta.env.TURSO_AUTH_TOKEN,
+});
+```
+
+## Querying your database
+
+To access information from your database, import `turso` and [execute a SQL query](https://docs.turso.tech/libsql/client-access/javascript-typescript-sdk#execute-a-single-statement) inside any `.astro` component.
+
+The following example fetches all `posts` from your table, then displays a list of titles in a `<BlogIndex />` component:
+
+```astro title="src/components/BlogIndex.astro"
+---
+import {turso} from '../turso'
+
+const {rows} = await turso.execute('SELECT * FROM posts')
+---
+
+<ul>
+  {rows.map((post) => (
+    <li>{post.title}</li>
+  ))}
+</ul>
+```
+
+### SQL Placeholders
+
+The `execute()` method can take [an object to pass variables to the SQL statement](https://docs.turso.tech/libsql/client-access/javascript-typescript-sdk#positional-placeholders), such as `slug`, or pagination.
+
+The following example fetches a single entry from the `posts` table `WHERE` the `slug` is the retrieved value from `Astro.params`, then displays **....**:
```

Comment:
I think all we need is a fill-in-the-blank here and I'm happy to merge!  🥳

Reply:
Awesome. Will resume this tomorrow first thing. Thank you for all your support getting this to the finish line.

Reply:
If you wanted to add something now you could update this to be “displays the title of the post” or “displays the title of the post that was retrieved using the slug”

Its almost midnight and on mobile so not great for suggestions 🙈